tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_HwenoModifiedSolution.cpp

```diff
@@ -0,0 +1,1290 @@
+      Element<3>::Neighbors_t{
+          {Direction<3>::lower_xi(), make_neighbor_with_id<3>(1)},
+          {Direction<3>::upper_xi(), make_neighbor_with_id<3>(2)},
+          {Direction<3>::lower_eta(), make_neighbor_with_id<3>(3)},
+          {Direction<3>::upper_eta(), make_neighbor_with_id<3>(4)},
+          {Direction<3>::lower_zeta(), make_neighbor_with_id<3>(5)},
+          {Direction<3>::upper_zeta(), make_neighbor_with_id<3>(6)}}};
+}
+
+void test_hweno_modified_solution_1d() {
+  INFO("Testing compute_hweno_modified_neighbor_solution in 1D");
+  using TagsList = tmpl::list<ScalarTag>;
+  const auto element = make_element<1>();
+  const auto mesh = Mesh<1>{
+      {{3}}, Spectral::Basis::Legendre, Spectral::Quadrature::GaussLobatto};
+  const auto logical_coords = logical_coordinates(mesh);
+
+  const auto primary_neighbor =
+      std::make_pair(Direction<1>::lower_xi(), ElementId<1>{1});
+  const auto upper_xi_neighbor =
+      std::make_pair(Direction<1>::upper_xi(), ElementId<1>{2});
+
+  const auto local_tensor = [&logical_coords]() noexcept {
+    const auto& x = get<0>(logical_coords);
+    return Scalar<DataVector>{{{1. - 0.2 * x + 0.4 * square(x)}}};
+  }
+  ();
+
+  const auto primary_vars = [&mesh, &logical_coords ]() noexcept {
+    Variables<TagsList> result(mesh.number_of_grid_points());
+    const auto x = get<0>(logical_coords) - 2.;
+    get(get<ScalarTag>(result)) = 4. - 0.5 * x - 0.1 * square(x);
+    return result;
+  }
+  ();
+
+  const auto upper_xi_vars = [&mesh]() noexcept {
+    // Large values to make sure this neighbor is excluded
+    Variables<TagsList> result(mesh.number_of_grid_points(), 1.e3);
+    return result;
+  }
+  ();
+
+  const auto make_tuple_of_means = [&mesh](
+      const Variables<TagsList>& vars) noexcept {
+    return tuples::TaggedTuple<::Tags::Mean<ScalarTag>>(
+        mean_value(get(get<ScalarTag>(vars)), mesh));
+  };
+
+  struct PackagedData {
+    tuples::TaggedTuple<::Tags::Mean<ScalarTag>> means;
+    Variables<TagsList> volume_data;
+    Mesh<1> mesh;
+  };
+  std::unordered_map<std::pair<Direction<1>, ElementId<1>>, PackagedData,
+                     boost::hash<std::pair<Direction<1>, ElementId<1>>>>
+      neighbor_data{};
+  neighbor_data[primary_neighbor].means = make_tuple_of_means(primary_vars);
+  neighbor_data[primary_neighbor].volume_data = primary_vars;
+  neighbor_data[primary_neighbor].mesh = mesh;
+  neighbor_data[upper_xi_neighbor].means = make_tuple_of_means(upper_xi_vars);
+  neighbor_data[upper_xi_neighbor].volume_data = upper_xi_vars;
+  neighbor_data[upper_xi_neighbor].mesh = mesh;
+
+  Scalar<DataVector> modified_tensor;
+  SlopeLimiters::compute_hweno_modified_neighbor_solution<ScalarTag>(
+      make_not_null(&modified_tensor), local_tensor, element, mesh,
+      neighbor_data, primary_neighbor);
+
+  // The expected coefficient values for the result of the constrained fit are
+  // found using Mathematica, using the following code:
```

Comment:
:+1: for inlining the mathematica to generate. Please include a Mathematica version number in these, though.